### PR TITLE
Improve location tracking

### PR DESCRIPTION
**Pull request by vsaarinen**
We currently track the location of the user to store the currently "selected" project/branch into the Redux state tree. This was done by registering a callback to location changes and firing off a Redux action that stored selected project/branch. This had the undesirable effect of triggering a second render whenever we changed the location.

Simplify this by listening to `react-router-redux`'s LOCATION_CHANGE actions and move the parsing logic to the Selected component's reducer.

TODO:
- [x] Rebase this onto master once #227 has been merged.

### CODE REVIEW

File: test/reducers.spec.ts [hunk 1]
```diff
@@ -461,11 +462,11 @@ describe('reducers', () => {
     });
 
     it('it replaces existing selections', () => {
```
Comment (dennari) [inline test/reducers.spec.ts]: Extra 'it'

File: test/reducers.spec.ts [hunk 2]
```diff
@@ -10,7 +11,7 @@ import { FetchError } from '../src/js/modules/errors';
 import Modal, { ModalType } from '../src/js/modules/modal';
 import Projects, { ProjectState } from '../src/js/modules/projects';
 import Requests, { RequestsState } from '../src/js/modules/requests';
-import Selected, { SelectedState, SetSelectedAction } from '../src/js/modules/selected';
+import Selected, { SelectedState } from '../src/js/modules/selected';
 
 /*const testData = {
```
Comment (dennari) [inline test/reducers.spec.ts]: Could the commented testData be removed?

Comment (vsaarinen) [inline test/reducers.spec.ts]: ~All of the tests need to refactored. I'd rather do that when that time comes.

Comment (dennari) [inline test/reducers.spec.ts]: Check

**Pull request merged**